include/seqan3/search/detail/policy_search_result_builder.hpp

```diff
@@ -39,73 +39,81 @@ struct policy_search_result_builder
     static_assert(!std::same_as<search_result_type, empty_type>, "The search result type was not configured properly.");
 
     /*!\brief Returns all hits (index cursors) without calling locate on each cursor.
+     *
      * \tparam index_cursor_t The type of index cursor used in the search algorithm.
+     * \tparam query_index_t The index type of the query.
+     * \tparam callback_t The callback which is called for every hit.
+     *
      * \param[in] internal_hits internal_hits A range over internal cursor results.
-     * \returns a range over seqan3::search_result.
+     * \param[in] idx The index associated with the current query.
+     * \param[in] callback The callback to invoke for every hit.
      *
      * \details
      *
      * The result is independent from the search modus (all, single_best, all_best, strata).
      */
-    template <typename index_cursor_t>
-    auto make_results(std::vector<index_cursor_t> internal_hits)
+    template <typename index_cursor_t, typename query_index_t, typename callback_t>
+    auto make_results(std::vector<index_cursor_t> internal_hits, query_index_t idx, callback_t && callback)
```

Comment:
```suggestion
    void make_results(std::vector<index_cursor_t> internal_hits, query_index_t idx, callback_t && callback)
```
more expressive now?

also for the others if you agree